azurerm/internal/services/batch/batch_job.go

```diff
@@ -0,0 +1,318 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	batchDataplane "github.com/Azure/azure-sdk-for-go/services/batch/2020-03-01.11.0/batch"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/sdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/batch/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/batch/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type BatchJobResource struct{}
+
+var _ sdk.ResourceWithUpdate = BatchJobResource{}
+
+type BatchJobModel struct {
+	Name                     string            `tfschema:"name"`
+	BatchPoolId              string            `tfschema:"batch_pool_id"`
+	DisplayName              string            `tfschema:"display_name"`
+	Priority                 int               `tfschema:"priority"`
+	MaxTaskRetryCount        int               `tfschema:"max_task_retry_count"`
+	CommonEnvironmentSetting map[string]string `tfschema:"common_environment_setting"`
+}
+
+func (r BatchJobResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.JobName,
+		},
+		"batch_pool_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.PoolID,
+		},
+		"display_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"common_environment_setting": {
```

Comment:
```suggestion
		"common_environment_properties": {
```

Reply:
The `azurerm_batch_pool` used the term for [`environment`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/batch_pool#environment), shall we use that name to be consistent?

Reply:
i would rather use the name with the better UX and leave a comment to change the old one in v3.0